package/MDAnalysis/core/universe.py

```diff
@@ -87,12 +87,28 @@
 import numpy as np
 import logging
 import copy
-import uuid
 import warnings
 
 import MDAnalysis
 import sys
 
+# When used in an MPI environment, importing MDAnalysis may trigger an MPI
+# warning because importing the uuid module triggers a call to os.fork().
+# This happens if MPI_Init() has been called prior to importing MDAnalysis.
+# The problem is actually caused by the uuid module and not by MDAnalysis
+# itself. Python 3.7 fixes the problem. However, for Python < 3.7, the uuid
+# module works perfectly fine with os.fork() disabled during import.
+# A clean solution is therefore simply to disable os.fork() prior to
+# importing the uuid module and to re-enable it afterwards.
+if sys.version_info[0] + 0.1 * sys.version_info[1] >= 3.7:
+    import uuid
+else:
+    import os
+    _os_dot_fork, os.fork = os.fork, None
+    import uuid
+    os.fork = _os_dot_fork
+    del _os_dot_fork
```

Comment:
Don't you want to 
```python
del os
```
to remove it from the namespace?

Reply:
1. I think you are correct from the point of view that MDA should behave the same way regardless of the Python version used, so `del`'ing the `os` module from the local namespace would fix that.
2. The `os` module might potentially be used in `MDAnalysis.core.universe` at some later point in time. From that point of view, I consider removing whatever module from the local namespace at an arbitrary position in the code as a bad idea.

In order to satisfy both considerations, my suggestion is to import the `os` module regardless of the Python version:
```Python
import os
if sys.version_info >= (3, 7):
    import uuid
else:
    _os_dot_fork, os.fork = os.fork, None
    import uuid
    os.fork = _os_dot_fork
    del _os_dot_fork
```
Importing the `os` module shoudn't cause any overhead since it is imported at other places in the code anyway (e.g., in `MDAnalysis.core.groups`). What do you think?

Reply:
Yeah we're more than likely importing os somewhere else (considering we almost always load a file) so fix 2 looks fine